packages/odyssey-react-mui/src/Button.tsx

```diff
@@ -79,7 +79,23 @@ export type ButtonProps = {
    * The variant of the Button
    */
   variant: (typeof buttonVariantValues)[number];
-};
+} & (
+  | {
+      text: string;
+      ariaLabel?: string;
+      ariaLabelledBy?: string;
+    }
+  | {
+      text?: undefined | "";
+      ariaLabel: string;
+      ariaLabelledBy?: string;
+    }
+  | {
+      text?: undefined | "";
+      ariaLabel?: string;
+      ariaLabelledBy: string;
+    }
+);
```

Comment:
Ask Shred about situations where you have an icon-only button but no tooltip

Reply:
Our current guidance/intent is for all icon-only Buttons to be tooltipped. That way they're both consistent and provide an affordance. `aria-` attributes aren't as useful for users who aren't on an AST but still need context.